botorch/acquisition/analytic.py

```diff
@@ -120,6 +106,53 @@ def __init__(
             best_f = torch.tensor(best_f)
         self.register_buffer("best_f", best_f)
 
+    @classmethod
+    def construct_inputs(
+        cls,
+        model: Model,
+        training_data: TrainingData,
+        objective: Optional[AcquisitionObjective] = None,
+        **kwargs: Any
+    ) -> Dict[str, Any]:
+        r"""Construct kwargs for the acquisition function's constructor.
+
+        Args:
+            model: The model to be used in the acquisition function.
+            training_data: A TrainingData object contraining the model's
+                training data. Used to extract `best_f` input.
+            objective: The objective to in the acquisition function.
+
+        Returns:
+            A dict mapping kwarg names of the constructor to values.
+
+        Example:
+            >>> kwargs = ExpectedImprovement.construct_inputs(
+                    model, training_data,
+                )
+            >>> EI = ExpectedImprovement(**kwargs)
+        """
+        best_f = training_data.Y.max(-2).values.squeeze(-1)
```

Comment:
If using a multi-output model with a `ScalarizedObjective`, I think we need to apply the same scalarization here before taking the max.